test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -1281,6 +1293,49 @@ public void TensorFlowImageClassification()
                 Assert.Equal(1, metrics.MicroAccuracy);
                 Assert.Equal(1, metrics.MacroAccuracy);
             }
+
+            // Testing TrySinglePrediction: Utilizing PredictionEngine for single
+            // predictions.
+            var predictionEngine = mlContext.Model
+                .CreatePredictionEngine<ImageData, ImagePrediction>(loadedModel);
+
+            IEnumerable<ImageData> testImages = LoadImagesFromDirectory(
+                fullImagesetFolderPath, true);
+
+            string[] directories = Directory.GetDirectories(fullImagesetFolderPath);
+            string[] labels = new string[directories.Length];
+            for(int j = 0; j < labels.Length; j++)
+            {
+                var dir = new DirectoryInfo(directories[j]);
+                labels[j] = dir.Name;
+            }
+
+            ImageData imageToPredictFirst = new ImageData
+            {
+                ImagePath = testImages.First().ImagePath,
+                Label = testImages.First().Label
+            };
+
+            ImageData imageToPredictLast = new ImageData
+            {
+                ImagePath = testImages.Last().ImagePath,
+                Label = testImages.Last().Label
+            };
+
+            var predictionFirst = predictionEngine.Predict(imageToPredictFirst);
+            var predictionLast = predictionEngine.Predict(imageToPredictLast);
+
+            var labelColumnFirst = schema.GetColumnOrNull("Label").Value;
+            var labelTypeFirst = labelColumnFirst.Type;
+            var labelCountFirst = labelTypeFirst.GetKeyCount();
+            var labelColumnLast = schema.GetColumnOrNull("Label").Value;
+            var labelTypeLast = labelColumnLast.Type;
+            var labelCountLast = labelTypeLast.GetKeyCount();
+
+            Assert.Equal((int)labelCountFirst, predictionFirst.Score.Length);
+            Assert.Equal((int)labelCountLast, predictionLast.Score.Length);
+            Assert.True(Array.IndexOf(labels, predictionFirst.PredictedLabel) > -1);
+            Assert.True(Array.IndexOf(labels, predictionLast.PredictedLabel) > -1);
```

Comment:
Can you please add an assert that predictionFirst and predictionLast have the same value? and also put an assert on the value? (it should remain the same since your seed is fixed)

Reply:
I'm sorry I meant can you please assert on the value for predictionFirst.PredictedLabel and predictionLast.PredictedLabel ... instead of just checking its greater than -1

---
In reply to: [330779542](https://github.com/dotnet/machinelearning/pull/4228#discussion_r330779542) [](ancestors = 330779542)